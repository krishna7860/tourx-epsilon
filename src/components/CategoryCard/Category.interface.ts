interface CategoryCardProps {
  id: number;
  title: string;
  image: string;
  description: string;
}

export default CategoryCardProps;
