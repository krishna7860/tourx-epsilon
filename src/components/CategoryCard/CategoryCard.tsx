import React from "react";
import {
  Container,
  HeaderContainer,
  Image,
  ImageContainer,
  SubTitle,
  Title,
} from "./style";
import CategoryCardProps from "./Category.interface";

const CategoryCard = (props: CategoryCardProps): JSX.Element => {
  const { title, image, description } = props;
  return (
    <Container>
      <ImageContainer>
        <Image src={image} />
      </ImageContainer>
      <HeaderContainer>
        <Title>{title}</Title>
        <SubTitle>{description}</SubTitle>
      </HeaderContainer>
    </Container>
  );
};

export default CategoryCard;
