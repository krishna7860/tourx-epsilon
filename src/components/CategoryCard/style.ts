import styled from "styled-components";

export const Container = styled.div`
  padding: 0.75rem;
  width: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  cursor: pointer;
  :hover {
    transition: all 0.2s;
    background: #eaecee;
    border-radius: 12px;
  }
`;

export const ImageContainer = styled.div`
  width: 100%;
  height: 160px;
  border: none;
  border-radius: 12px;
  overflow: hidden;
`;

export const Image = styled.img`
  height: 100%;
  width: 100%;
  :hover {
    transition: all 0.2s ease-in-out;
    transform: scale(1.2);
  }
  :not(:hover) {
    transition: all 0.2s ease-in-out;
    transform: scale(1);
  }
`;

export const HeaderContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  align-self: flex-start;
  margin-top: 1rem;
`;

export const Title = styled.p`
  font-size: 18px;
  margin: 0px;
  color: black;
`;

export const SubTitle = styled.p`
  font-size: 14px;
  margin: 0px;
  color: #808b96;
`;
