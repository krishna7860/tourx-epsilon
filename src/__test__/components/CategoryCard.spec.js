/* eslint-disable react/jsx-filename-extension */
/* eslint-disable import/extensions */
import React from "react";
import { render } from "@testing-library/react";
import CategoryCard from "../../components/CategoryCard/CategoryCard.tsx";

describe("components/categoryCard", () => {
  it("should be rendered properly", () => {
    const { asFragment } = render(
      <CategoryCard
        image="https://res.cloudinary.com/thrillophilia/image/upload/c_fill,dpr_1.5,f_auto,fl_progressive.strip_profile,g_center,h_183,q_auto,w_245/v1/filestore/ivniu6zyafkwdxaggg7jl7msro47_27023582_1061778550630758_383729866060832850_o.jpg"
        title="Hotel"
        description="Sample description"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
