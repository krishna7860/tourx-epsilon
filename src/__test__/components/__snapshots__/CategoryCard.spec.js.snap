// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/categoryCard should be rendered properly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM bpwbdL"
  >
    <div
      class="sc-gtsrHT jiMVHD"
    >
      <img
        class="sc-dlnjwi htgYQL"
        src="https://res.cloudinary.com/thrillophilia/image/upload/c_fill,dpr_1.5,f_auto,fl_progressive.strip_profile,g_center,h_183,q_auto,w_245/v1/filestore/ivniu6zyafkwdxaggg7jl7msro47_27023582_1061778550630758_383729866060832850_o.jpg"
      />
    </div>
    <div
      class="sc-hKFxyN dXkjxw"
    >
      <p
        class="sc-eCApnc dSgpmJ"
      >
        Hotel
      </p>
      <p
        class="sc-jSFjdj eKXNkF"
      >
        Sample description
      </p>
    </div>
  </div>
</DocumentFragment>
`;
